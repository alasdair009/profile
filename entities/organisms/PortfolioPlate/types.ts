import { ContentPlateProps } from "@/entities/organisms/ContentPlate/ContentPlate.types";
import { LinkProps } from "@/entities/atoms/Link/types";
import { HTMLAttributes } from "react";

export type PortfolioPlateProps = {
  /**
   * Props applied to the ContentPlate
   */
  contentPlateProps: Omit<ContentPlateProps, "children">;
  /**
   * Heading for the piece
   */
  heading: string;
  /**
   * Url to the piece
   */
  url?: LinkProps["href"];
} & HTMLAttributes<HTMLDivElement>;
