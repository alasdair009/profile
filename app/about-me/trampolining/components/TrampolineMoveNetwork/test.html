<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
