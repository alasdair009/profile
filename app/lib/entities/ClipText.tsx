"use client";

import styled from "styled-components";
import {rem} from "polished";

export const ClipText = styled.span`
  align-items: center;
  //background: black;
  display: flex;
  font-size: ${rem(56)};
  height: 100%;
  justify-content: center;
  left: 50%;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
  width: 100%;
  -webkit-text-fill-color: transparent;
  text-fill-color: transparent;
  -webkit-background-clip: text;
  background-clip: text;
  background: black;
  color: white;
`;