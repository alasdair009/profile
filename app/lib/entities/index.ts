"use client";

import styled from "styled-components";

export * from "@/entities";
export { Splash } from "./Splash";

export const FireWrapper = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  position: absolute;
  width: 100%;
`;
export const FirePadding = styled.div<{ $backgroundColor: string }>`
  background: ${({ $backgroundColor }) => $backgroundColor};
  flex: 1;
`;
