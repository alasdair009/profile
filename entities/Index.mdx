import { Meta } from "@storybook/blocks";
import Image from "next/image";
import amLogo from "./assets/am.svg";
import { rem } from "polished";

<Meta title="Alasdair Macrae Design" />

# Alasdair Macrae Design

<Image
  src={amLogo}
  alt="Green and white AM letters"
  width={0}
  height={0}
  style={{ maxWidth: rem(240), width: "100%", height: "auto" }}
/>

I am a keen supported of Storybook js using it across the majority of projects I build from websites, web-apps and even desktop applications.
