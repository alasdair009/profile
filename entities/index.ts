export { borderRadii } from "./design-tokens/effects";
export { colors } from "./design-tokens/colors";
export { fonts } from "./design-tokens/typography";
export { sizes } from "./design-tokens/dimensions";

export { Button } from "./atoms/Button";
export { Fire } from "./atoms/Fire";
export { Heading } from "./atoms/Heading";
export { Link } from "./atoms/Link";
export { Orb } from "./atoms/Orb";
export { Paragraph } from "./atoms/Paragraph";
export { Spacer } from "./atoms/Spacer";
export { TextMask } from "./atoms/TextMask";

export { Container } from "./organisms/Container";
export { ContentPlate } from "./organisms/ContentPlate";
export { Header } from "./organisms/Header";
export { Splash } from "./organisms/Splash";
