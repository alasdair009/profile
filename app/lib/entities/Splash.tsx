import {HTMLAttributes} from "react";
import styled from "styled-components";
import {globalContentMaxWidth} from "@/entities/design-tokens/dimensions";
import {rem} from "polished";
import {Header, Heading, Paragraph} from "@/entities";

type SplashProps = {
} & HTMLAttributes<HTMLDivElement>;

const Root = styled.section`
  height: calc(100vh - ${Header.height});
  margin: 0 auto;
  max-width: ${rem(globalContentMaxWidth)};
  position: relative;
  width: 100%;
`;

const SplashCopy = styled.div`
  left: 50%;
  max-width: 100%;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
`;

export function Splash({...rest}: SplashProps) {
    return <Root {...rest}>
        <SplashCopy>
            <Heading>Alasdair Macrae</Heading>
            <Paragraph align="center" fontSize="large"><em>Web Developer and Trampolinist</em></Paragraph>
        </SplashCopy>
    </Root>
}