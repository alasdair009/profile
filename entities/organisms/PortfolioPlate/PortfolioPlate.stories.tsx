import { Meta, StoryObj } from "@storybook/react";
import { PortfolioPlate } from "./PortfolioPlate";
import jdsLogo from "../../assets/jds-logo.svg";
import jagexLauncherImage from "../../assets/launcher.png";
import { colors, Link, Paragraph } from "@/entities";
import { expect, within } from "@storybook/test";

const meta: Meta<typeof PortfolioPlate> = {
  component: PortfolioPlate,
  argTypes: {
    children: {
      control: { type: "text" },
    },
  },
};
export default meta;

export const Default: StoryObj<typeof PortfolioPlate> = {
  args: {
    contentPlateProps: {
      foregroundImage: jdsLogo,
      foregroundImageAlt: "White Jagex logo part-way through being drawn",
      backgroundCss: `linear-gradient(#1c1c1c,${colors.blackEvil})`,
    },
    heading: "JDS",
    children: (
      <>
        <Paragraph>
          Responsible for constructing a large design system to style and
          maintain consistent branding across all Jagex Publishing Platform
          products including websites and apps. This library was written in
          React and displayed via Storybook.js.
        </Paragraph>
        <Paragraph>
          The system covers all areas of visual implementation including
          colours, animation, typography, sizing, components and is all
          structured using atomic design principles.
        </Paragraph>
      </>
    ),
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const linkElement = canvas.queryByTestId(Link.name);
    const headingElement = canvas.getByText(args.heading);

    await expect(linkElement).not.toBeInTheDocument();
    await expect(headingElement).toBeInTheDocument();
  },
};

export const WithUrl: StoryObj<typeof PortfolioPlate> = {
  args: {
    contentPlateProps: {
      foregroundImage: jagexLauncherImage,
      foregroundImageAlt:
        "The opening UI from the Jagex Launcher displaying RuneScape",
      backgroundCss: `linear-gradient(#07111b,${colors.blackEvil})`,
    },
    heading: "Jagex Launcher",
    url: "https://www.jagex.com/launcher",
    children: (
      <Paragraph>
        The Jagex Launcher is an application that is used to run all Jagex
        products. I was responsible for maintaining the{" "}
        <abbr title="Jagex Design System">JDS</abbr> (see above) integration
        that delivered the visual aesthetics and functionality for the
        front-end.
      </Paragraph>
    ),
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const linkElement = canvas.getByTestId(Link.name);
    const headingElement = canvas.getByText(args.heading);

    await expect(linkElement).toBeInTheDocument();
    await expect(linkElement).toHaveAttribute("href", args.url);
    await expect(headingElement).toBeInTheDocument();
  },
};
