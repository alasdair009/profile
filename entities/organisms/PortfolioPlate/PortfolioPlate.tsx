import { ContentPlate, Heading, Link } from "@/entities";
import { PortfolioPlateProps } from "./types";

export function PortfolioPlate({
  contentPlateProps,
  heading,
  url,
  children,
  ...rest
}: PortfolioPlateProps) {
  return (
    <ContentPlate
      {...contentPlateProps}
      data-testid={PortfolioPlate.name}
      {...rest}
    >
      <Heading level="h3">{heading}</Heading>
      {url ? (
        <Link href={url} />
      ) : !contentPlateProps.embedUrl ? (
        <pre>(URL not public)</pre>
      ) : (
        <></>
      )}
      {children}
    </ContentPlate>
  );
}
