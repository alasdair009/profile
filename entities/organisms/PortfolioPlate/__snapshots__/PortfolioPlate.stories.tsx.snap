// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms/PortfolioPlate Default play-test 1`] = `
<section data-testid="ContentPlate"
         class="sc-crribH bXfjUx"
>
  <figure class="sc-gpSXuZ hLiPLT">
  </figure>
  <div class="sc-kuTxxw gZGEOM">
    <div class="sc-blcnQh enYcrv">
      <h3 data-testid="Heading"
          class="sc-hfvVTD cmyMfV"
      >
        JDS
      </h3>
      <pre>
        (URL not public)
      </pre>
      <p data-testid="Paragraph"
         class="sc-gLXSEc fjewFw"
      >
        Responsible for constructing a large design system to style and maintain consistent branding across all Jagex Publishing Platform products including websites and apps. This library was written in React and displayed via Storybook.js.
      </p>
      <p data-testid="Paragraph"
         class="sc-gLXSEc fjewFw"
      >
        The system covers all areas of visual implementation including colours, animation, typography, sizing, components and is all structured using atomic design principles.
      </p>
    </div>
    <figure class="sc-bHSjBK dUnWvr">
      <img alt="White Jagex logo part-way through being drawn"
           loading="lazy"
           width="350"
           height="333"
           decoding="async"
           data-nimg="1"
           class="sc-iSnsYj hHkmHU"
           srcset="/static/media/entities/assets/jds-logo.svg?w=384&amp;q=75 1x, /static/media/entities/assets/jds-logo.svg?w=750&amp;q=75 2x"
           src="/static/media/entities/assets/jds-logo.svg?w=750&amp;q=75"
           style="color: transparent;"
      >
    </figure>
  </div>
</section>
`;

exports[`organisms/PortfolioPlate WithUrl play-test 1`] = `
<section data-testid="ContentPlate"
         class="sc-crribH bXfjUx"
>
  <figure class="sc-gpSXuZ fijHNR">
  </figure>
  <div class="sc-kuTxxw gZGEOM">
    <div class="sc-blcnQh enYcrv">
      <h3 data-testid="Heading"
          class="sc-hfvVTD cmyMfV"
      >
        Jagex Launcher
      </h3>
      <a target="_blank"
         rel="noopener"
         data-testid="Link"
         class="sc-eEPDDI gughtU"
         href="https://www.jagex.com/launcher"
      >
        https://www.jagex.com/launcher
        <img title="Third party site, opens in a new tab."
             alt="Icon to represent this opens in a new tab"
             loading="lazy"
             width="128"
             height="128"
             decoding="async"
             data-nimg="1"
             class="sc-gutikT BEgvF"
             srcset="/static/media/entities/assets/new-tab-icon.svg?w=128&amp;q=75 1x, /static/media/entities/assets/new-tab-icon.svg?w=256&amp;q=75 2x"
             src="/static/media/entities/assets/new-tab-icon.svg?w=256&amp;q=75"
             style="color: transparent;"
        >
      </a>
      <p data-testid="Paragraph"
         class="sc-gLXSEc fjewFw"
      >
        The Jagex Launcher is an application that is used to run all Jagex products. I was responsible for maintaining the
        <abbr title="Jagex Design System">
          JDS
        </abbr>
        (see above) integration that delivered the visual aesthetics and functionality for the front-end.
      </p>
    </div>
    <figure class="sc-bHSjBK dUnWvr">
      <img alt="The opening UI from the Jagex Launcher displaying RuneScape"
           loading="lazy"
           width="1280"
           height="717"
           decoding="async"
           data-nimg="1"
           class="sc-iSnsYj hHkmHU"
           srcset="/static/media/entities/assets/launcher.webp?w=1920&amp;q=75 1x, /static/media/entities/assets/launcher.webp?w=3840&amp;q=75 2x"
           src="/static/media/entities/assets/launcher.webp?w=3840&amp;q=75"
           style="color: transparent;"
      >
    </figure>
  </div>
</section>
`;
